v3/lints/rfc/lint_name_constraint_not_fqdn.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package rfc
+
+import (
+	"strings"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+type nameConstraintNotFQDN struct{}
+
+func (l *nameConstraintNotFQDN) Initialize() error {
+	return nil
+}
+
+func (l *nameConstraintNotFQDN) CheckApplies(c *x509.Certificate) bool {
+	return util.IsExtInCert(c, util.NameConstOID)
+}
+
+func (l *nameConstraintNotFQDN) Execute(c *x509.Certificate) *lint.LintResult {
+
+	var incorrectPermittedHosts []string
+	var incorrectExcludedHosts []string
+	var errString string
+
+	incorrectPermittedHosts = collectNotFQDNEntries(c.PermittedURIs)
+	incorrectExcludedHosts = collectNotFQDNEntries(c.ExcludedURIs)
+
+	if len(incorrectPermittedHosts) != 0 {
+		errString += buildErrorString(incorrectPermittedHosts, true)
+	}
+	if len(incorrectPermittedHosts) != 0 && len(incorrectExcludedHosts) != 0 {
+		errString += "; "
+	}
+	if len(incorrectExcludedHosts) != 0 {
+		errString += buildErrorString(incorrectExcludedHosts, false)
+	}
+
+	if len(errString) != 0 {
+		return &lint.LintResult{
+			Status:  lint.Error,
+			Details: errString,
+		}
+	}
+
+	return &lint.LintResult{Status: lint.Pass}
+}
+
+func collectNotFQDNEntries(hosts []x509.GeneralSubtreeString) []string {
+	var incorrectHosts []string
+
+	for _, subtreeString := range hosts {
+		host := subtreeString.Data
+
+		host = strings.TrimPrefix(host, ".")
+
+		if !util.IsFQDN(host) {
+			incorrectHosts = append(incorrectHosts, host)
+		}
+	}
+
+	return incorrectHosts
+}
```

Comment:
I'm sorry, but I'm still not 100% sure if i fully understood your original idea. This function does more or less what you described in your previous comment, but instead of the []string it takes a []x509.GeneralSubtreeString. I could write another function that collects the data from this list and return a list of strings too, but I didn't think it was necessary. Does this look good to you now?